src/gluonts/mx/trainer/_base.py

```diff
@@ -115,19 +118,44 @@ def __init__(
         batch_size: int = 32,
         num_batches_per_epoch: int = 50,
         learning_rate: float = 1e-3,
-        learning_rate_decay_factor: float = 0.5,
-        patience: int = 10,
-        minimum_learning_rate: float = 5e-5,
+        learning_rate_decay_factor: Optional[int] = None,
+        patience: Optional[int] = None,
+        minimum_learning_rate: Optional[int] = None,
         clip_gradient: float = 10.0,
         weight_decay: float = 1e-8,
         init: Union[str, mx.initializer.Initializer] = "xavier",
         hybridize: bool = True,
-        avg_strategy: Union[
-            AveragingStrategy, IterationAveragingStrategy
-        ] = SelectNBestMean(num_models=1),
-        post_initialize_cb: Optional[Callable[[mx.gluon.Block], None]] = None,
+        callbacks: Union[CallbackList, List[Callback]] = CallbackList(
```

Comment:
Minor: since `CallbackList` is a `Callback`, this could take `Union[Callback, List[Callback]]`. This way one can pass a single callback without any wrapping.

Reply:
good idea, thanks! done